main.py

```diff
@@ -1,78 +1,31 @@
-from enum import Enum
-import argparse
-
+import typer
 import prediction_market_agent as pma
-from prediction_market_agent.tools.types import xdai_type
+from decimal import Decimal
+from prediction_market_agent.agents.all_agents import AgentType, get_agent
+from prediction_market_agent.tools.types import xDai, Mana
 from prediction_market_agent.tools.utils import should_not_happen, check_not_none
+from prediction_market_agent.markets.all_markets import MarketType, get_binary_markets
 
 
-class AgentType(Enum):
-    LANGCHAIN = 1
-    AUTOGEN = 2
-    ALWAYS_YES = 3
-    LLAMAINDEX = 4
-    METAGPT = 5
-    CREWAI = 6
-    CUSTOM_OPENAI = 7
-    CUSTOM_LLAMA = 8
-
-
-class Market(Enum):
-    MANIFOLD = 1
-    OMEN = 2
-
-
-agent_mapping = {
-    AgentType.LANGCHAIN: pma.agents.langchain_agent.LangChainAgent,
-    AgentType.AUTOGEN: pma.agents.autogen_agent.AutoGenAgent,
-    AgentType.ALWAYS_YES: pma.agents.always_yes.AlwaysYesAgent,
-    AgentType.LLAMAINDEX: pma.agents.llamaindex_agent.LlamaIndexAgent,
-    AgentType.METAGPT: pma.agents.metagpt_agent.MetaGPTAgent,
-    AgentType.CREWAI: pma.agents.crewai_agent.CrewAIAgent,
-    AgentType.CUSTOM_OPENAI: pma.agents.custom_agent.CustomAgent.init_with_openai,
-    AgentType.CUSTOM_LLAMA: pma.agents.custom_agent.CustomAgent.init_with_llama,
-}
-
-pick_binary_market_mapping = {
-    Market.MANIFOLD: pma.manifold.pick_binary_market,
-    Market.OMEN: pma.omen.pick_binary_market,
-}
-
-if __name__ == "__main__":
-    args = argparse.ArgumentParser()
-    args.add_argument(
-        "--agent-type",
-        type=str,
-        choices=[t.name.lower() for t in list(AgentType)],
-        default=AgentType.ALWAYS_YES.name.lower(),
-    )
-    args.add_argument(
-        "--market",
-        type=str,
-        choices=[t.name.lower() for t in list(Market)],
-        default=Market.MANIFOLD.name.lower(),
-    )
-    args.add_argument(
-        "--auto-bet",
-        type=bool,
-        default=False,
-        help="If true, does not require user input to place the bet.",
-    )
-
-    parsed_args = args.parse_args()
-    selected_market = Market[parsed_args.market.upper()]
-    agent_type = AgentType[parsed_args.agent_type.upper()]
+def main(
```

Comment:
I changed to typer to simplify the code; hope you will like it! 

![Screenshot by Dropbox Capture](https://github.com/evangriffiths/prediction-market-agent/assets/15619339/b4b39ce9-a4a6-4262-990c-d2890e1257b3)